base/strings/substring.jl

```diff
@@ -123,32 +119,13 @@ function unsafe_convert(::Type{Ptr{R}}, s::SubString{String}) where R<:Union{Int
     convert(Ptr{R}, pointer(s.string)) + s.offset
 end
 
-## reversed strings without data movement ##
-
-struct RevString{T<:AbstractString} <: AbstractString
-    string::T
-end
-
-endof(s::RevString) = endof(s.string)
-length(s::RevString) = length(s.string)
-sizeof(s::RevString) = sizeof(s.string)
-
-function next(s::RevString, i::Int)
-    n = endof(s); j = n-i+1
-    (s.string[j], n-prevind(s.string,j)+1)
-end
-
 """
-    reverse(s::AbstractString) -> AbstractString
-
-Reverses a string.
+    reverse(s::AbstractString) -> String
```

Comment:
I'd keep `AbstractString` (or remove any mention of the return type), since an implementation e.g. for `UTF32String` would rather return a string of the same type.

Reply:
Yes, sorry, I changed this before I realized that `reverse` needs to return the same type.